controllers/auth.js

```diff
@@ -0,0 +1,41 @@
+
+
+const { StatusCodes } = require('http-status-codes')
+const { BadRequestError } = require('../errors')
+const User = require('../models/User');
+
+
+// Register a new user
+// POST api/auth/register
+const registerUser = async (req, res) => {
+    const { name, email, password } = req.body;
+    const user = await User.create({ name, email, password });
+    res.status(StatusCodes.CREATED).json({ user })
```

Comment:
By returning the whole user object, you're exposing their password. Even if it's encrypted, exposing the user password is not a good idea. You could return just the name and email.

```suggestion
    res.status(StatusCodes.CREATED).json({ name, email })
```